### DISCUSSION TITLE
[BUG]SM wireless changes speed switches to both = faster

### DISCUSSION THREAD

**Discussion by WidescreenEnt** [microdee/OpenUnrealSpaceMouse#46] [Q&A] [answered]
Hi,

First, thanks for making the plugin.  FWIW, I like yours much better than 3DC's.

I ran across a bug with the Space Mouse Wireless and UE4.26: 

PREP: I made sure that camera speed faster/slower were set to buttons 1 and 0 in the Editor Preferences.  

STEPS: - Return to viewport
-Make sure Camera Speed indicator is at a speed > 1
- Press 0 (left) button -> Camera Speed indicator decreases by 1 (Success!)
- Press 1 (right) button -> Camera Speed indicator does not change (? Expected behavior?)
- Press 1 (right) button again -> Camera Speed indicator increases by one (Success!)
- Press 0 (left) button -> Camera Speed indicator does not change (? Expected behavior?)
- Press 0 (left) button again -> Camera Speed indicator decreases by one (Success!)
- Press 1 (right) button -> Camera Speed indicator does not change (? Expected behavior?)
- Press 1 (right) button again -> Camera Speed indicator increases by one (Success!)

Here it comes--

- Press 0 (left) button -> Camera Speed indicator does not change (? Expected behavior?)
- Press 0 (left) button again -> Camera Speed indicator INCREASES by one (FAIL!)

Now BOTH buttons are assigned to increase Camera Speed.
Oddly, opening SM settings in Editor Preferences shows both functions assigned to button 0.  (If you run the steps in reverse, starting with button 1, you end up with both functions assigned to button 1.)

Please fix this, as it interferes with the use of the SpaceMouse.

Let me know if you need additional data.  Thanks for looking at this.

**Comment by microdee** [comment, marked as answer]
thanks for the report. can you please send a screenshot of your button binding setting? I have a suspicion you haven't unticked button learning checkboxes (yeah making that more trivial to use is on my backlog)

**Reply by WidescreenEnt** [reply]
You're 100% correct.  I misread the docs.  I thought those were activation checkboxes.  They are unchecked and everything is working correctly now.  Thanks!

BTW, are buttons 2 and 3 supported on the SM Wireless?  Perhaps through chording or multiple clicks?  Here are my default key bindings:

![image](https://user-images.githubusercontent.com/80491688/112885208-53729380-9085-11eb-8c18-b50b88365042.png)

**Reply by microdee** [reply]
hah, that was a good hunch. for your other question: unfortunately no, that device has only 2 buttons, so you can only assign 2 distinct actions to those buttons. I don't have short term plans for supporting button press gestures or puck pushing gestures.

**Reply by WidescreenEnt** [reply]
No worries.  I'm delighted that the SpaceMouse works as well as it does.  Thanks again!
